rfc-106-docker-for-local-development.md

```diff
@@ -0,0 +1,41 @@
+# RFC 106: Use Docker for local development
+
+## Summary
+
+Adopt a Docker-based approach for local development instead of the Vagrant VM.
+
+## Problem
+
+We currently rely on the Vagrant development VM for the day to day dev cycle.
+
+- It has [many documented issues](https://docs.publishing.service.gov.uk/manual.html#development-vm)
+- Changes to puppet often result in a broken VM, because the infrastructure team don't use it.
+- We are adopting more cloud-based services into our stack, like S3 and Amazon's Elasticsearch. This means that our development environment doesn't look like production anymore, but we still need code in govuk-puppet.
+- It's hard to keep updated. Running the puppet command will often error and take a long time.
+- GOV.UK is moving to a containerised infrastructure
+
+## Possible approaches to local development
+
+- An approach based on **Docker and docker-compose**. Ben Thorner has a working prototype called [govuk-docker](https://github.com/benthorner/govuk-docker) and we have the [end-to-end tests](https://github.com/alphagov/publishing-e2e-tests) running in the same. Other GDS programmes like GOV.UK Pay use this approach.
```

Comment:
If we are eventually using k8s in production... does using docker-compose introduce another set of config to manage?

Reply:
GSP have been looking at how to run [k8s locally](https://github.com/alphagov/gsp/blob/master/docs/gds-supported-platform/getting-started-gsp-local.md), and there's also a tool to [migrate from docker compose to k8s](https://github.com/kubernetes/kompose)

Reply:
The GSP uses [Helm Charts](https://helm.sh/docs/developing_charts/) this is a collection of yaml that describes the application so that it can be deployed to a kubernetes cluster. 

As @erino mentioned kompose may be useful